### PR TITLE
feature(unlock-app) - verifiers use token and login fix

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
- Fix components and endpoint issue
- Better error message based on different states

Also, attach one video that shows all the functionalities

https://user-images.githubusercontent.com/20865711/171603437-0fbc7587-245b-4638-8462-a20c535f7fd6.mov

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by kalidiagne** [comment]
@searchableguy can you please check what cause `SyntaxError: Unexpected token < in JSON at position 0` on endpoints also with status 200?

**Comment by searchableguy** [comment]
Which endpoint?

**Comment by kalidiagne** [comment]
> Which endpoint?

Oh my bad, there is a typo on the endpoint, all fine!

**Review by searchableguy — APPROVED** [review]
LGTM. We could refactor this after we have a few more functions using the new endpoints.

File: unlock-app/src/services/storageService.ts [hunk 1]
```diff
@@ -47,23 +47,72 @@ interface GetSiweMessageProps {
...
       host,
     })
+    this.accessToken = null
   }
 
   async login(message: string, signature: string) {
     return this.locksmith.login(message, signature)
   }
 
+  private setToken(token: string) {
```
Comment (searchableguy) [inline unlock-app/src/services/storageService.ts:76]: Remove the private. if the function can be used externally, just keep it as it is.

File: unlock-app/src/services/storageService.ts [hunk 2]
```diff
@@ -47,23 +47,72 @@ interface GetSiweMessageProps {
...
+    const expireAt: number = decoded?.exp ?? -1
+    if (decoded && expireAt) {
+      const startTime = new Date().getTime()
+      const expireTime = new Date(expireAt).getTime()
+      const timeout = (startTime - expireTime) / 1000 / 60 // time difference in seconds
+      console.table({
+        startTime,
+        expireTime,
+        timeout,
+      })
```
Comment (searchableguy) [inline unlock-app/src/services/storageService.ts:88]: Why the frontend log? Remove.

File: unlock-app/src/components/content/VerifiersContent.tsx [hunk 3]
```diff
@@ -56,16 +55,22 @@ export const VerifiersContent: React.FC<VerifiersContentProps> = ({
...
-            Authorization: `Bearer ${token}`,
           },
         }
         await storageService
           .getEndpoint(
-            `/verifier/${network}/${lockAddress}/${resolvedAddress}`,
-            options
+            `/v2/api/verifier/${network}/${lockAddress}/${resolvedAddress}`,
+            options,
+            true
```
Comment (julien51) [inline unlock-app/src/components/content/VerifiersContent.tsx:64]: Can you add a quick comment as to what this `true` is for?

**Review by julien51 — APPROVED** [review]
Please do a bit of cleanup and then let's merge it!

File: unlock-app/src/components/content/VerifiersContent.tsx (hunk 3 above)
Comment (julien51) [inline unlock-app/src/components/content/VerifiersContent.tsx:64]:
```suggestion
            true /* withAuth */
```

File: unlock-app/src/components/interface/verifiers/VerifiersList.tsx [hunk 4]
```diff
@@ -137,11 +126,17 @@ export const VerifiersList: React.FC<VerifiersListProsps> = ({
         <tbody>
           {verifiers?.map((verifier: any, index: number) => {
             const key = `${verifier?.address}-${index}`
+            const createdAt = verifier?.createdAt
+              ? new Date(verifier?.createdAt).toLocaleString()
+              : '-'
+            const verifiedAt = verifier?.verifiedAt
```
Comment (julien51) [inline unlock-app/src/components/interface/verifiers/VerifiersList.tsx:132]: What is this? I don't think we need it here.

Comment (kalidiagne) [reply unlock-app/src/components/interface/verifiers/VerifiersList.tsx:132]: is `updatedAt`

**Pull request merged**